File: src/com/javascript-practice/linkList/isUnivalueList.js
class Node {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

const isUnivalueList = (head, prev = null) => {
    if (head == null) return true;

    if (prev !== null && prev !== head.val) return false;
    return isUnivalueList(head.next, head.val)
};

const a = new Node(7);
const b = new Node(7);
const c = new Node(4);

a.next = b;
b.next = c;

// 7 -> 7 -> 7

console.log(isUnivalueList(a)); // true

File: src/com/structy/LinkedList/IsUnivalueList.java
package com.structy.LinkedList;

public class IsUnivalueList {

    public boolean isUnivalueList(Node head){
        if(head == null) return true;
        int firstVal = head.val;
        Node current = head.next;
        while (current!=null){
            int nextVal = current.val;
            if(firstVal!=nextVal){
               return false;
            }
           current = current.next;
        }
        return true;
    }
    public boolean isUnivalueListRec(Node head){

        return helper(head, head.val);
    }
    private boolean helper(Node head, int firstVal){
        if(head == null) return true;
        if(firstVal != head.val)
            return false;
        else
            return helper(head.next, firstVal);
    }
    public static void main(String[] args) {
        Node a = new Node(2);
        Node b = new Node(1);
        Node c  = new Node(1);
        a.next = b;
        b.next = c;
        IsUnivalueList isUnivalueList = new IsUnivalueList();
        System.out.println(isUnivalueList.isUnivalueList(a));
    }
}
